tensorboard/data/server_ingester.py

```diff
@@ -92,13 +102,19 @@ def start(self):
         else:
             reload = str(int(self._reload_interval))
 
+        sample_hint_pairs = [
+            "%s=%s" % (k, v) for k, v in self._samples_per_plugin.items()
+        ]
+        samples_per_plugin = ",".join(sample_hint_pairs)
+
         args = [
             server_binary,
             "--logdir=%s" % (self._logdir,),
             "--reload=%s" % reload,
             "--port=0",
             "--port-file=%s" % (port_file_path,),
             "--die-after-stdin",
+            "--samples-per-plugin=%s" % samples_per_plugin,
```

Comment:
I’m noticing that while this works fine with the data server at head, it
will fail for the v0.3.0 data server (no such flag), which is what’s
currently released. What do you think about either:

  - only including this flag when `samples_per_plugin` is non-empty, as
    you had previously? then `--load_fast` would still be broken with
    v0.3.0 with explicit `--samples_per_plugin`, but at least that’s not
    in the default path; or

  - explicitly gating the flag on `tensorboard_data_server.__version__`?
    you could use `pkg_resources.parse_version` to ensure that it’s at
    least `0.4.0a0` (cf. `version_test.py`)

I appreciate that this kind of dependency constrant is less than ideal,
and can think about ways to improve it in the general case.

Reply:
> only including this flag when samples_per_plugin is non-empty, as you had previously?

Done.

Agreed that it's non ideal, but hopefully the `--undefok` Clap may go through before we do the next release of `tensorboard-data-server`.

Explicitly gating the flag on version # would work for us, but I'm not sure what the story would be for users writing their own custom data server using it via `TENSORBOARD_DATA_SERVER_BINARY`.

Besides `--undefok`, the only thing I have thought of is trying to detect support for the flag by running `<binary> --help` and parsing the output to see whether `--samples-per-plugin` is there.  Which is also very non-ideal!

Reply:
> Explicitly gating the flag on version # would work for us, but I'm not sure what the story would be for users writing their own custom data server using it via `TENSORBOARD_DATA_SERVER_BINARY`.

Ah, good point, yes. You could invoke `rustboard --version` and see what
it prints, which is not *too* hard but certainly less convenient.

> Besides `--undefok`, the only thing I have thought of is trying to detect support for the flag by running `<binary> --help` and parsing the output to see whether `--samples-per-plugin` is there. Which is also very non-ideal!

Also a good point. Certainly not ideal, and only really works for
feature additions (e.g., would be harder to detect if we add support for
something like `--samples-per-plugin=scalars=100,images=all`).
But I don’t mind it that much if we want to go that route.